viewing_party/party.py

```diff
@@ -1,23 +1,131 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    new_movie ={}
+    if not title or not genre or not rating:
+        return None
+    else:    
+        new_movie["title"] = title 
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data 
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total = 0
+    if not user_data["watched"]:
+        average_rating = 0.0
+        return average_rating
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    average_rating = total/len(user_data["watched"])
+    return average_rating
 
+def get_most_watched_genre(user_data):
+    genres =[]
+    count = 0
+    
+    if not user_data["watched"]:
+        return None

```

Comment:
Let's move the `genres` and `count` creation after the guard clause. 

```suggestion
    if not user_data["watched"]:
        return None
    genres = []
    count = 0
```